internal/shard/shard_thread.go

```diff
@@ -28,6 +28,7 @@ type ShardThread struct {
 	store            *dstore.Store                      // store that the shard is responsible for.
 	ReqChan          chan *ops.StoreOp                  // ReqChan is this shard's channel for receiving requests.
 	workerMap        map[string]chan *ops.StoreResponse // workerMap maps workerID to its unique response channel
+	preProcessingMap map[string]chan *ops.StoreResponse // map to maintain list of preprocessing requests
```

Comment:
Maybe we should also use this space to explain what a pre-processing request is and why it's needed.

Reply:
Removed the extra map